### ISSUE TITLE
ExpressionPrinter and MethodCallExpression

### ISSUE THREAD

**Issue by dmitry-lipetsk** [aspnet/EntityFrameworkCore#13683]
EF expression:

```c#
var recs=db.testTable.Where(r => !(r.DATA+null).Equals("ASDFGH123qwerty") && r.TEST_ID==testID);
```

Log:

> Compiling query model: 
> 'from TEST_RECORD r in DbSet<TEST_RECORD>
> where !(bool [r].DATA + null.Equals("ASDFGH123qwerty")) && [r].TEST_ID == (Nullable<long>)__testID_0
> 

For me - "bool **[r].DATA + null**.Equals("ASDFGH123qwerty")" looks incorrectly.

I think, "bool **([r].DATA + null)**.Equals("ASDFGH123qwerty")" looks better.

Could you correct the ExpressionPrinter.VisitMethodCall for do it?

**Comment by maumar** [comment]
fixed in e3112e595a34b7ae93b6297f43562300b2e67f8b

**Comment by dmitry-lipetsk** [comment]
@maumar,

Could you look to this printer output:

> Compiling query model: 
> 'from TEST_RECORD r in DbSet<TEST_RECORD>
> where (Nullable<int>)[r].DATA + __vv2_0.Length == null && [r].TEST_ID == __testID_1
> select [r]'
 
Source linq experession:
`c#
  var recs=db.testTable.Where(r => (r.DATA+vv2).Length==null && r.TEST_ID==testID);
`

**Question**
The following query looks better or no:
> Compiling query model: 
> 'from TEST_RECORD r in DbSet<TEST_RECORD>
> where (Nullable<int>)([r].DATA + __vv2_0.Length) == null && [r].TEST_ID == __testID_1
> select [r]'
?

**Comment by maumar** [comment]
It does look better to me. Problem is, we don't want to add to many parentheses. Perhaps good heuristic is to add them if you have convert whose operand is a binary expression

**Comment by maumar** [comment]
fixed in f54a7241e78c9d9c8d87e80ec7cd68a1526767da
